test/Quickstart.kt

```diff
@@ -0,0 +1,198 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+// We need to access private function Studio.MainWindow, this allows us to.
+// Do not use this outside of tests anywhere. It is extremely dangerous to do so.
+@file:Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER")
+
+package com.vaticle.typedb.studio.test
+
+import androidx.compose.ui.test.assertAll
+import androidx.compose.ui.test.hasClickAction
+import androidx.compose.ui.test.junit4.ComposeContentTestRule
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onAllNodesWithText
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import com.vaticle.typedb.client.TypeDB
+import com.vaticle.typedb.client.api.TypeDBOptions
+import com.vaticle.typedb.client.api.TypeDBSession
+import com.vaticle.typedb.client.api.TypeDBTransaction
+import com.vaticle.typedb.studio.Studio
+import com.vaticle.typedb.studio.framework.common.WindowContext
+import com.vaticle.typedb.studio.state.StudioState
+import com.vaticle.typeql.lang.TypeQL
+import com.vaticle.typeql.lang.query.TypeQLMatch
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import org.junit.Rule
+import org.junit.Test
+import java.io.File
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Paths
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+/**
+ * Some of these tests use delay!
+ *
+ * The rationale for this is that substituting in stub classes/methods would create a lot of friction from release to
+ * release as the tests would require updating to completely reflect all the internal state that changes with each
+ * function. As a heavily state-driven application, duplicating all of this functionality and accurately verifying that
+ * the duplicate is like-for-like is out of scope.
+ *
+ * The delays are:
+ *  - used only when necessary (some data is travelling between the test and TypeDB)
+ *  - generous with the amount of time for the required action.
+ *
+ * However, this is a source of non-determinism and a better and easier way may emerge.
+ */
+class Quickstart {
+    @get:Rule
+    val composeRule = createComposeRule()
+
+    // This test simulates the carrying out of the instructions found at https://docs.vaticle.com/docs/studio/quickstart
+    @Test
+    fun `Quickstart`() {
```

Comment:
This looks like a pretty solid E2E test! However, its implementation is very "flat", which makes it hard to see at a glance what it actually does.

To rectify this, we should break down the method into submethods with suitable names, like "connectToTypeDB", "validateAnswers" .... etc.

Reply:
Good shout, looks a lot better now. This has been implemented.